### PR TITLE
Peons should not report SysMonitor stats since MiddleManager reports them.

### PR DESCRIPTION
**Pull request by tejaswini-imply**
### Description

`Sysmonitor` stats (mem, fs, disk, net, cpu, swap, sys, tcp) are reported by all Druid processes, including _Peons_ that are ephemeral in nature. Since _Peons_ always run on the same host as the _MiddleManager_ that spawned them and is unlikely to change, the SyMonitor metrics emitted by _Peon_ are merely duplicates. This is often not a problem except when machines are super-beefy. Imagine a 64-core machine and 32 workers running on this machine. now you will have each _Peon_ reporting metrics for each core. that's an increase of (32 * 64)x in the number of metrics. This leads to a metric explosion.

This PR updates `MetricsModule` to check node role running while registering `SysMonitor` and not to load any existing `SysMonitor$Stats`.

<hr>

##### Key changed/added classes in this PR
 * `MetricsModule`
 * `SysMonitor`

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

File: core/src/test/java/org/apache/druid/java/util/metrics/SysMonitorTest.java
Comment (tejaswini-imply) [reply core/src/test/java/org/apache/druid/java/util/metrics/SysMonitorTest.java:30]: A separate annotation can be created to disable these tests on arm64-graviton2 architecture conditionally.

File: server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java
Comment (suneet-s) [inline server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java:186]: Instead of leaking `isPeon` into the SysMonitor - have you considered introducing a new class `NoopSysMonitor` that does nothing. Then in this provider  if the role is a Peon, return the noop class.

I think that would be a cleaner division of responsibilities and it would make it easier to test as well.

Comment (tejaswini-imply) [reply server/src/main/java/org/apache/druid/server/metrics/MetricsModule.java:186]: I completely agree although I was a little skeptical to use that if any extra functionality might be added in SysMonitor in the future that might want to be used on Peons. But that's a long shot. I'll go with `NoopSysMonitor`. Thanks.

**Comment by capistrant** [comment]
This is a great find and fix. Looks like CI is upset about coverage on the new code, but I will be a +1 once that is resolved

File: core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java
Comment (suneet-s) [inline core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: nit: I think it should be safe for this test to run on all architectures, but I guess there's no harm in skipping this test on aarch64

```suggestion
```

**Comment by suneet-s** [comment]
I've retriggered what appears to have been flaky test failures.

+1 from me, I'll leave it to other committers in this thread to approve and merge this change.

Thanks for the contribution @tejaswini-imply !

File: core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java
Comment (tejaswini-imply) [reply core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: This test fails on aarch64 since SysMonitor is the parent of NoopSysMonitor. Sigar still gets loaded even though there's no need for it. Or should there be another class that these two classes inherit from to avoid this?

Comment (suneet-s) [reply core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: Oops I didn't notice that when I skimmed the change. The NoopSysMonitor should just implement Monitor since it is doing nothing. Or you can extend AbstractMonitor and return false in the doMonitor() method.

Comment (tejaswini-imply) [reply core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: But `MetricsModule` is getting the instance of `SysMonitor` through Guice while loading, and `getSysMonitor()` provides for SysMonitor. In case of peon returning NoopSysMonitor here.

Comment (suneet-s) [reply core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: ah ok. feel free to ignore my suggestion above.

There is a way to refactor this so that sigar won't initialize on the peons, but it seems like overkill.

Comment (tejaswini-imply) [reply core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: I thought of adding logic in `getMonitorScheduler` during loading for the same and it seemed overkill. If you don't mind could you please share if there is other way for future references.

**Review by abhishekagarwal87 — APPROVED** [review]

**Comment by tejaswini-imply** [comment]
Thanks @abhishekagarwal87, @suneet-s for the review.

File: core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java
Comment (suneet-s) [reply core/src/test/java/org/apache/druid/java/util/metrics/NoopSysMonitorTest.java:35]: If we wanted to do something like that, I would have made SysMonitor an interface and make the current class an implementation of that interface.

Another way to approach this would be to add a new annotation and inject the Monitor interface that way https://github.com/google/guice/wiki/BindingAnnotations

**Pull request merged**
